Section 1: Introduction and Setup/4-RoR Development Environment/testing_project/app/assets/config/manifest.js

```diff
@@ -0,0 +1,2 @@
+//= link_tree ../images
```

Comment:
what is this file for?

Reply:
`manifest.js` was created by default when running `rails new` which specify which files to be used as a top-level target using sprockets methods `link`, `link_directory`, and `link_tree`. 
In the file above, `link_tree` would include all files found inside the `/images` directory as well as its sub-directories while `link_directory` would only include all files in the parent folder, excluding files inside the sub-directories.

Reply:
Don't understand what you are saying. Layman language please

Reply:
Starting from Rails 4 and above, `app/assets/config/manifest.js` is created on default when running `rails new`. The `manifest.js` file is needed for Sprockets used in Rails for compiling and serving web assets such as JS files, CSS files and images etc. (From Rails 6 onwards, Webpack is used for compiling JS files on default)

Inside `manifest.js`, using Sprockets directive, we could configure assets that we wanted to include during compilation for our app.

For example:
```
//= link ../javascript/application.js
//= link_directory ../stylesheets .css
//= link_tree ../images
```
`link`: include only specific file
In the example above, this would include only single file `application.js`

`link_directory`: include all files excluding the subfolders in this path
In the example above, this would include all files ending with `.css` inside the `stylesheets` directory

`link_tree`: include all files and files inside subfolders in this path
In the example above, this would include all files and files inside subfolders inside the `image` directory

Reply:
Ok. You say its needed for sprockets. Why & how is it used by sprockets?

Reply:
Why?
Sprockets is a Ruby library for compiling and serving web assets. Sprockets allows to organize an application’s JavaScript and CSS files into smaller more manageable chunks that can be distributed over a number of directories and files. It provides structure and practices on how to include assets in our projects.

How?
Using directives inside `manifest.js` file, Sprockets can determine which files are included during compilation. When it comes to deploying your application, Sprockets then uses these directives to turn your multiple files into a single file for better performance.